### PR TITLE
LPD-2161 Prevent XSS vulnerability by applying regex in RichText source mode

### PR DESCRIPTION
**Pull request by danvb112**
Related Issues: 
https://liferay.atlassian.net/browse/LPD-2161
https://liferay.atlassian.net/browse/LPD-16314

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ const INITIAL_EDITING_LOCALE = {
 	localeId: themeDisplay.getDefaultLanguageId(),
 };
 
+const ALERT_REGEX = /alert\((.*?)\)/;
+const INNER_HTML_REGEX = /innerHTML\s*=\s*.*?/;
+const ON_ERROR_REGEX = /onerror\s*=\s*.*?/;
+const ON_LOAD_REGEX = /onload\s*=\s*.*?/;
+const ON_MOUSE_OVER_REGEX = /onmouseover\s*=\s*.*?/;
+
```
Comment (igor-franca) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:34]: Hey @danvb112, As mentioned by you, let's check if using a library like Dom Purify can make things easier here. let me know when we get a response from FE-Infra

**Comment by igor-franca** [comment]
@danvb112, Furthermore, the problem in SF is related to your commit message.

Jenkins console: 

> The commit message contains the word 'xss', which could reveal potential security vulnerablities.

**Comment by danvb112** [comment]
ci:test:sf

**Comment by igor-franca** [comment]
Waiting for direction [from](https://liferay.slack.com/archives/CNBG06JS3/p1707240730380119)

**Comment by danvb112** [comment]
Hey @igor-franca, we will follow this approach, can you please review it?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js [hunk 2]
```diff
@@ -189,10 +195,12 @@ const RichText = ({
...
-								const sanitizedValue = value.replace(
-									/onerror="[^"]+"/gi,
-									''
-								);
+								const sanitizedValue = value
+									.replace(ON_ERROR_REGEX, '')
+									.replace(ALERT_REGEX, '')
+									.replace(INNER_HTML_REGEX, '')
+									.replace(ON_LOAD_REGEX, '')
+									.replace(ON_MOUSE_OVER_REGEX, '');
```
Comment (igor-franca) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:203]: Hi @danvb112, after some investigation I realized that the infinite loop was not caused by `escapeHTML` in `frontend-js`.

The root cause was probably changing event.data.dataValue inside the `onSetData` function, triggering the function repeatedly.

I found relevant [documentation](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_best_practices.html#security) that might be useful to us.

This documentation shows when we can sanitize user input to mitigate XSS attack using the `contentPreview` function. We can test to see if we can use `escapeHTML` in `contentPreview`.

**Review by igor-franca — CHANGES_REQUESTED** [review]

**Review by igor-franca — APPROVED** [review]
Performing the tests with malicious inputs, I was unable to execute any malicious javascript code.

Furthermore, I didn't encounter any unexpected behavior. Great Job @danvb112!

**Comment by danvb112** [comment]
ci:test:forms

**Comment by danvb112** [comment]
ci:test:data-engine

**Comment by igor-franca** [comment]
> The ci:test:data-engine are ok 👍🏾 
> The ci:test:forms are ok 👍🏾

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
ci:test:stable

**Comment by igor-franca** [comment]
> The ci:test:relevant are ok 👍🏾

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**